src/uct/sm/mm/mm_posix.c

```diff
@@ -269,30 +420,44 @@ static ucs_status_t uct_posix_free(void *address, uct_mm_id_t mm_id, size_t leng
         goto err;
     }
 
-    file_name = ucs_malloc(NAME_MAX, "shared mr posix mmap");
+    file_name = ucs_calloc(1, NAME_MAX, "shared mr posix mmap");
     if (file_name == NULL) {
         ucs_error("Failed to allocate memory for the shm_unlink file name. %m");
         status = UCS_ERR_NO_MEMORY;
         goto err;
     }
 
-    /* use the mmid (63 bits uuid) to recreate the file_name for shm_unlink */
-    sprintf(file_name, "/ucx_shared_mr_uuid_%zu", mm_id >> 1);
-    if (shm_unlink(file_name) != 0) {
-        ucs_warn("unable to unlink the shared memory segment. File name is: %s",
-                 file_name);
-        status = UCS_ERR_SHMEM_SEGMENT;
+    status = uct_posix_set_path(file_name, mm_id & UCT_MM_POSIX_SHM_OPEN, path,
+                                mm_id >> UCT_MM_POSIX_CTRL_BITS);
+    if (status != UCS_OK) {
         goto err_free_file;
     }
 
+    /* use the mmid (62 bits uuid) to recreate the file_name for unlink */
+    if (mm_id & UCT_MM_POSIX_SHM_OPEN) {
+        if (shm_unlink(file_name) != 0) {
+            ucs_warn("unable to unlink the shared memory segment. File name is: %s",
+                     file_name);
+            status = UCS_ERR_SHMEM_SEGMENT;
+        }
+    } else {
+        if (unlink(file_name) != 0) {
+            ucs_warn("unable to unlink the shared memory segment %m. File name is: %s",
+                     file_name);
+            status = UCS_ERR_SHMEM_SEGMENT;
+        }
+    }
+
 err_free_file:
```

Comment:
is this error flow or exit flow?
if exit flow - should be called 'out_free_file'